test cases/cpp/1 import std/meson.build.clang

```diff
@@ -0,0 +1,42 @@
+project(
+  'import std',
+  'cpp',
+  version : '0.1',
+  meson_version : '>= 1.3.0',
+  default_options : ['warning_level=3', 'cpp_std=c++23'],
+)
+
+
+cpp_comp = find_program('clang++', dirs: '/usr/local/Cellar/llvm/20.1.1/bin/')
+
+cpp20_std_module = custom_target(
+  'cpp20-std-module',
+  command: [
+    cpp_comp,
+    '-std=c++23',
+    '-Wno-reserved-identifier',
+    '-Wno-reserved-module-identifier',
+    '--precompile',
+    '-o',
+    '@OUTPUT@',
+    '@INPUT@',
+  ],
+  input: '/usr/local/Cellar/llvm/20.1.1/share/libc++/v1/std.cppm',
+  output: 'std.pcm',
+)
+
+cpp20_std_module_dep = declare_dependency(
+  compile_args: ['-fmodule-file=std=@0@'.format(cpp20_std_module.full_path())],
```

Comment:
None of this should be in the build files. These need to be handled in the background. For comparison, precompiled headers work by just adding `cpp_pch: 'foo/bar.h'`, everything else is done behind the scenes. The "actual code" that does things is in `ninjabackend.py`.

I don't have a clear idea what the API should actually be, but maybe something along the lines of:

```meson
project('foo', 'cpp', 
    default_options: ['cpp_std=c++26', 'cpp_modules=std'])
executable('importstd', 'importstd.cpp')
```

Reply:
This is how it works (I made some changes based on feedback from the chat in the internals of how it works, but externally it goes like this):

```
project(
  'import std',
  'cpp',
  version : '0.1',
  meson_version : '>= 1.3.0',
  default_options : ['warning_level=3', 'cpp_std=c++23',
                     'cpp_import_std=true'],
)


exe = executable(
  'import std',
  'import_std.cpp'
)
```